features_module_how_to_share_sites_config_with_oth.md

```diff
@@ -0,0 +1,123 @@
+
+One possible solution is to document all the configurations made and repeat them in the new environment. However, this is not a very clever way of working. It takes a lot of time and there are too many opportunities to make mistakes along the way. 
+
+Another possibility is to manually export/import configuration in code, as you can see from the example in View under the options:
+
+![](images/features/export_view.png)
+
+Views, Content types, Panels, and some other modules provide this export/import option.  However, not all of modules gives us this option. In addition, working this way doesn't provide us with the ability to know the dependencies of our configuration. For example, when we export View, we need to export also the content type it depends on, otherwise the View won't work. And when the View is complex, there can be more and more configurations that it relies on. How can we be sure we won't miss anythings?
+
+Our solution and the most efficient option, is to use the Features module!
+
+Features simply packages up all of that configuration into a module, and we can use it like any other Drupal module. In another words: **Features is a module that creates modules.**
+
+
+## Creating a feature
+
+Let's use a simple example - a web page that displays a list of blog post. 
+
+![](images/features/blog_page.png)
+
+
+It was built from a content type called `Blog`, and a View.
+Now we want to export our work and move it to another environment. We can simply create a module (using features module) that contain the blog configurations (Content type and View). 
+
+Let's do it together:
+
+Go to `admin/structure/features/create`.
+Give the feature a name. This will be the name of the module, so we always use the project name and a reasonable name that tell us what this module use for (Blog in this case). Also write a description that tells more about this module.
+In the right area on the screen you need to select the components that should be include in the module (in this case Content type=Blog and View=Blog).
+As you can see, Features automatically select for us all the dependencies our module needs to work properly. You can add your own dependencies as you want.
+
+![](images/features/create_feature.png)
+
+When you done, just click on `Download feature`.
+Uncompress the file, and put the module folder under `modules/custom` in the project directory.
+Enable the module so Drupal can start using it.
+
+**Note**: When you make your configurations in DB, then export it as a feature and enable the module, your configurations stored both in code and DB. In that case, we can't be sure our module works exactly how we expected (try to disable the module and you still get the same configuration). The most effective way to make sure all our configuration moved successfully to code, is to start new installation and then enable the module.
```

Comment:
@liatsade  This is not clear from the second sentence and I need your help to fix it up. I think stating something about not being sure if you are seeing/working on the DB or code will be more clear. You explained ii great in the training but its not coming through in this paragraph.